### PR TITLE
OVN-K: Runbook for NoRunningOvnMaster alert

### PR DESCRIPTION
**Pull request by martinkennelly**
Signed-off-by: Martin Kennelly <mkennell@redhat.com>

/cc bmeng
/cc npinaeva

### CODE REVIEW

**Comment by martinkennelly** [comment]
This alert is currently severity warning but will be upgraded to a critical severity alert.

**Comment by npinaeva** [comment]
/lgtm
Great work!

File: alerts/cluster-network-operator/NoRunningOvnMaster.md
Comment (ravitri) [inline alerts/cluster-network-operator/NoRunningOvnMaster.md:9]: Does 10m seem too high without a running master? I think this might be a more severe condition than other critical alerts of OVN so could make it smaller perhaps to alert it sooner? My reference here is with [etcdNoLeader](https://github.com/openshift/cluster-etcd-operator/blob/master/manifests/0000_90_etcd-operator_03_prometheusrule.yaml#L32-L42) alert which would be fired after 1m. We might want to keep it to be 1-3 minutes. Let me know your thoughts on it.

Comment (ravitri) [reply alerts/cluster-network-operator/NoRunningOvnMaster.md:9]: Oh ok there's a separate `NoOvnMasterLeader` alert too. And I missed that the CNO would go degraded here and not down so perhaps 10m would be ok too.

File: alerts/cluster-network-operator/NoRunningOvnMaster.md
Comment (ravitri) [inline alerts/cluster-network-operator/NoRunningOvnMaster.md:72]: Would it be possible to cover some possible issue scenarios and solution(s) for it? Having references to some of the acceptable and safe solutions will be really helpful when the incident response team itself is changing from time to time. Covering most typical scenarios might suffice and we can add more later on when we come across it.

File: alerts/cluster-network-operator/NoRunningOvnMaster.md
Comment (bmeng) [inline alerts/cluster-network-operator/NoRunningOvnMaster.md:40]: This could be merged as 
```
oc logs deployment/network-operator -n openshift-network-operator
```

File: alerts/cluster-network-operator/NoRunningOvnMaster.md
Comment (bmeng) [inline alerts/cluster-network-operator/NoRunningOvnMaster.md:52]: maybe change the order? to check the ds firstly? As the master count is not quite a big problem usually.

File: alerts/cluster-network-operator/NoRunningOvnMaster.md
Comment (bmeng) [inline alerts/cluster-network-operator/NoRunningOvnMaster.md:48]: seems this section should be moved to below under the `ovn-kubernetes master pod`?

File: alerts/cluster-network-operator/NoRunningOvnMaster.md
Comment (bmeng) [inline alerts/cluster-network-operator/NoRunningOvnMaster.md:55]: Better to have some pre-condition here, like if you are not seeing the running ovn-kube master pod matches the expected daemonset replica count. And you can start the debugging from the problem ovn-kube master pod.

File: alerts/cluster-network-operator/NoRunningOvnMaster.md
Comment (bmeng) [reply alerts/cluster-network-operator/NoRunningOvnMaster.md:9]: Hi @ravitri There is another review[1] for those alerts, like when it should be triggered. And Martin suggested to reduce this from 10m to 1m.
One concern of mine, as if we are setting the trigger hold time too short. It might be mis-fired when the master node reboot takes longer. WDYT?

[1]https://github.com/orgs/openshift/teams/sre-alert-sme/discussions/2

Comment (martinkennelly) [reply alerts/cluster-network-operator/NoRunningOvnMaster.md:9]: This alert will only fire if _all_ the OVN-K master pods are not running. 1m is probably too short an interval and maybe 10m is too large, but until we have data for a correct time, we can leave it for now.
Please voice your opinion on this in the google doc in the link bo attached.

File: alerts/cluster-network-operator/NoRunningOvnMaster.md
Comment (martinkennelly) [reply alerts/cluster-network-operator/NoRunningOvnMaster.md:48]: yeah, I put it in the CNO section because CNO controls the ds. I moved it to ovn-k section.

File: alerts/cluster-network-operator/NoRunningOvnMaster.md
Comment (martinkennelly) [reply alerts/cluster-network-operator/NoRunningOvnMaster.md:72]: I have never seen all the pods fail so I am unsure which scenario may cause it besides guessing things like internet connectivity pulling image or API server connectivity but they are scenarios that should not be added here.
I think we can revisit this when it actually occurs and add it later.

File: alerts/cluster-network-operator/NoRunningOvnMaster.md
Comment (ravitri) [reply alerts/cluster-network-operator/NoRunningOvnMaster.md:9]: Yeah, agreed that 1m is too short too.. I was thinking about 3 minutes but then I agree with Martin. We need more data to decide better so can revisit it in future. Resolving this comment :+1:

File: alerts/cluster-network-operator/NoRunningOvnMaster.md
Comment (ravitri) [reply alerts/cluster-network-operator/NoRunningOvnMaster.md:71]: Ack.. Main intent was to have clear actionable steps for any alert.. We can revisit this too in future..

File: alerts/cluster-network-operator/NoRunningOvnMaster.md
Comment (wanghaoran1988) [inline alerts/cluster-network-operator/NoRunningOvnMaster.md:58]: `oc get logs` -> `oc logs` ?

Comment (martinkennelly) [reply alerts/cluster-network-operator/NoRunningOvnMaster.md]: Thank you! resolved.

File: alerts/cluster-network-operator/NoRunningOvnMaster.md
Comment (npinaeva) [inline alerts/cluster-network-operator/NoRunningOvnMaster.md:29]: Maybe first check if the status is ok with `oc get co network` and only if not go further with CNO?

**Comment by martinkennelly** [comment]
@npinaeva @bmeng I want your /lgtm again. All comments addressed.

**Comment by martinkennelly** [comment]
Once you merge this, I can then take control of the other PRs. Thanks Jan.

**Comment by martinkennelly** [comment]
@jan--f gentle reminder about this. Thank you.

**Review by jan--f — APPROVED** [review]
/lgtm

**Pull request merged**
